File: back_end/back_end/internal/model/enum.go
package model

import (
	"time"

	"gorm.io/gorm"
)

// 数据库实体相关类型
type ContestId uint
type ProblemId uint
type SubmissionId uint
type TagId uint
type UserId uint
type TagClassifyId uint

// 配置文件相关类型
type UserLangId string
type CodeLangId string
type VerdictId string
type ProblemLabel string

type BaseModel struct {
	ID        uint           `gorm:"primaryKey" json:"id"`
	CreatedAt time.Time      `json:"createdAt"`
	UpdatedAt time.Time      `json:"updatedAt"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`
}

File: front_end/front_end/src/interface/entity/enum.ts
// 存放在数据库中
export type ContestId = number
export type ProblemId = number
export type SubmissionId = number
export type TagId = number
export type TagClassifyId = number
export type UserId = number

// 配置文件，存放在 JSON 文件中
export type UserLangId = string
export type CodeLangId = string
export type VerdictId = string

export type ProblemLabel = string

export interface BaseModel {
  createdAt?: Date
  updatedAt?: Date
  deletedAt?: Date
}